docs/book/features/vsphere_csi_migration.md

```diff
@@ -0,0 +1,175 @@
+| Raw Block Volume | Supported | Not supported |
+| Inline volumes in Pod spec | Supported | Not supported |
+
+In addition to the above feature comparison, one of the most important things customers need to consider is that Kubernetes will deprecate In-tree vSphere volume plugin and it will be removed in the future Kubernetes releases.
+Volumes provisioned using vSphere in-tree plugin do not get the additional new features supported by the vSphere CSI driver.
+
+Kubernetes has provided a seamless procedure to help migrate in-tree vSphere volumes to a vSphere CSI driver. After in-tree vSphere volumes migrated to vSphere CSI driver, All subsequent operations on migrated volumes are performed by the vSphere CSI driver.
+Migrated vSphere volume will not get additional capabilities vSphere CSI driver supports.
+
+## Consider followings before turning on Migration
+
+- vSphere CSI Migration is released with `beta` feature-gate in Kubernetes 1.19. Refer [release note announcement](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.19.md#csi-migration---azuredisk-and-vsphere-beta).
+- Kubernetes 1.19 release has deprecated vSAN raw policy parameters for the in-tree vSphere Volume plugin and these parameters will be removed in a future release. Refer [deprecation announcement](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.19.md#deprecation)
+- Following vSphere in-tree StorageClass parameters will not be supported after enabling migration.
+  - `hostfailurestotolerate`
+  - `forceprovisioning`
+  - `cachereservation`
+  - `diskstripes`
+  - `objectspacereservation`
+  - `iopslimit`
+  - `diskformat`
+- Storage Policy should not be renamed or deleted after enabling migration. Volume migration requires storage policy to be present to successfully register volume as Container Volume in vSphere.
+- Datastore should not be renamed after enabling migration. Volume migration relies on the datastore name retrieved from the volume path available in the Persistent Volume Source.
+- For statically created vSphere in-tree Persistent Volume Claims and Persistent Volumes, make sure to add following annotations. Statically provisioned in-tree vSphere volumes can't be migrated to CSI without following these annotations.
+
+    Annotation on PV
+
+      annotations:
+        pv.kubernetes.io/provisioned-by: kubernetes.io/vsphere-volume
+
+    Annotation on PVC
+
+      annotations:
+        volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/vsphere-volume
+
+- vSphere CSI Driver does not support Provisioning `eagerzeroedthick` and `zeroedthick` volume. After the migration is enabled, when a new volume is requested using the in-tree provisioner and `diskformat` parameter, `diskformat` parameter is dropped by vSphere CSI Driver.
```

Comment:
we are blocking creation of such volumes right?

Reply:
yes. https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/common/util.go#L243-L246

made it clear

> - vSphere CSI Driver does not support Provisioning `eagerzeroedthick` and `zeroedthick` volume. After the migration is enabled, when a new volume is requested using the in-tree provisioner and `diskformat` parameter, volume creation will be failed by vSphere CSI Driver.

Reply:
@SandeepPissay do we want to allow creating volume when diskformat is setup to `thin` and only fail for `eagerzeroedthick` and `zeroedthick`?

Reply:
checked with @SandeepPissay 

we can allow creating volume if we observe `diskformat` is `thin. I will make a follow-up change.

Reply:
tracking issue here - https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/378

Reply:
new support statement is as below

> - vSphere CSI Driver does not support Provisioning `eagerzeroedthick` and `zeroedthick` volume. After the migration is enabled, when a new volume is requested using the in-tree provisioner and `diskformat` parameter set to `eagerzeroedthick` or `zeroedthick`, volume creation will be failed by vSphere CSI Driver. Post migration only supported value for `diskformat` parameter will be `thin`. Existing volumes created before the migration using disk format `eagerzeroedthick` or `zeroedthick` will be migrated to CSI.